### PR TITLE
RHINENG-4837: Add dumb-init to enable hbi processes using PID 1

### PR DESCRIPTION
**Pull request by thearifismail**
# Overview

This PR is being created to address [RHINENG-4837](https://issues.redhat.com/browse/RHINENG-4837).
@RonnyPfannschmidt here is the PR for your review.  

With the use of `ENTRYPOINT [ "dumb-init", ]`, now the HBI process run with `PID 1`
```
$ k exec -it host-inventory-mq-pmin-6869b8df94-cpxll -- top

top - 22:53:39 up 46 min,  0 users,  load average: 0.34, 0.48, 0.66
Tasks:   2 total,   1 running,   1 sleeping,   0 stopped,   0 zombie
%Cpu(s):  3.2 us,  1.4 sy,  0.0 ni, 94.2 id,  1.1 wa,  0.0 hi,  0.1 si,  0.1 st
MiB Mem :  15995.9 total,   2702.1 free,   4734.1 used,   8559.7 buff/cache
MiB Swap:      0.0 total,      0.0 free,      0.0 used.  10352.0 avail Mem 

    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND                                                                                
      1 1001      20   0 1717820  82560  19340 S   0.3   0.5   0:10.93 inv_mq_service.                                                                        
    564 1001      20   0   52144   4172   3452 R   0.0   0.0   0:00.00 top                                                                                    

$ k exec -it host-inventory-service-b5498db47-5pqlp -- top

top - 22:55:19 up 48 min,  0 users,  load average: 0.67, 0.51, 0.65
Tasks:   6 total,   1 running,   5 sleeping,   0 stopped,   0 zombie
%Cpu(s):  2.3 us,  2.3 sy,  0.0 ni, 94.3 id,  1.1 wa,  0.0 hi,  0.0 si,  0.0 st
MiB Mem :  15995.9 total,   2679.2 free,   4746.2 used,   8570.5 buff/cache
MiB Swap:      0.0 total,      0.0 free,      0.0 used.  10330.5 avail Mem 

    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND                                                                                
      1 1001      20   0  350228  70784  14452 S   0.0   0.4   0:01.43 gunicorn                                                                               
      8 1001      20   0 1203212  77488  13592 S   0.0   0.5   0:07.69 gunicorn                                                                               
      9 1001      20   0 1276948  77424  13508 S   0.0   0.5   0:06.87 gunicorn                                                                               
     10 1001      20   0 1277460  79244  14416 S   0.0   0.5   0:06.44 gunicorn                                                                               
     11 1001      20   0 1276948  77496  13592 S   0.0   0.5   0:06.68 gunicorn                                                                               
    213 1001      20   0   52144   4232   3508 R   0.0   0.0   0:00.00 top                                                                                    
```

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: deploy/clowdapp.yml [hunk 1]
```diff
@@ -138,9 +138,7 @@ objects:
       minReplicas: ${{REPLICAS_PMIN}}
       podSpec:
         command:
```
Comment (RonnyPfannschmidt) [inline deploy/clowdapp.yml:140]: This needs to use args,else it's broken on k8s

**Review by RonnyPfannschmidt — CHANGES_REQUESTED** [review]
Please adopt args and the entry point script as the issue mentions

Without the script and args everything is worse

File: deploy/clowdapp.yml (hunk 1 above)
Comment (RonnyPfannschmidt) [reply deploy/clowdapp.yml:140]: Command replaces the docker entry point

Args replaces the docker command

When using command the docker entry point is ignored

**Comment by thearifismail** [comment]
Replaced by https://github.com/RedHatInsights/insights-host-inventory/pull/1642

**Comment by kruai** [comment]
Closing this because it was replaced by another PR.

**Pull request closed without merging**